### ISSUE TITLE
CultureData initialization brings a lot of dependencies

### ISSUE THREAD

**Issue by marek-safar** [dotnet/runtime#46141]
CultureData [initialization](https://github.com/dotnet/runtime/blob/6072e4d3a7a2a1493f514cdf4be75a3d56580e84/src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.Unix.cs#L16) brings a lot of dependencies for members which are rarely used. Few examples are [_SpecificCulture](https://github.com/dotnet/runtime/blob/6072e4d3a7a2a1493f514cdf4be75a3d56580e84/src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.Unix.cs#L70) and [_bNeutral](https://github.com/dotnet/runtime/blob/6072e4d3a7a2a1493f514cdf4be75a3d56580e84/src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.Unix.cs#L68) but there are more.

It'd decrease the size of a trimmed app if we moved all these data into delayed initialization. It'd also help with code readability as today it's a mix of random jumps that can interfere with each other and are hard to follow.

**Comment by tarekgh** [comment]
@marek-safar could you point at _bNeutral data you are referring to?

I understand the benefit of that but I sated to worry the maintenance of the code started to be very complicated. so we need to weigh that against the benefit.

**Comment by marek-safar** [comment]
> could you point at _bNeutral data you are referring to?

https://github.com/dotnet/runtime/blob/496bc88f0517990c8430ba8d8d63946a6f9e716c/src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.cs#L1237

> worry the maintenance of the code started to be very complicated

I think the changes could actually lead to clearer/simple code

**Comment by tarekgh** [comment]
@marek-safar You are not pointing at _bNeutral at all. would be good to fix the description of the issue.

**Comment by marek-safar** [comment]
What do you mean? The link in the description points to a line where _bNeutral is set and when you asked for data I added another comment point to value _bNeutral is set with.

**Comment by tarekgh** [comment]
Maybe I am missing something here. the line initialize `_bNeutral` is 

https://github.com/dotnet/runtime/blob/496bc88f0517990c8430ba8d8d63946a6f9e716c/src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.cs#L789

which is not really using any data. the link you sent is pointing at 

```C#
internal string TwoLetterISOCountryName => _sISO3166CountryName ??= GetLocaleInfoCore(LocaleStringData.Iso3166CountryName); 
```

which I have no idea how this is related to `_bNeutral`.

**Comment by marek-safar** [comment]
Your code is I believe for invariant mode only, if you click on the link in the description it should take you to https://github.com/dotnet/runtime/blob/6072e4d3a7a2a1493f514cdf4be75a3d56580e84/src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.Unix.cs#L68

**Comment by tarekgh** [comment]
@eerhardt I moved this to 7.0 milestone. Let me know if you still think we need to have for 6.0 and @joperezr is planning to fix it?

**Comment by eerhardt** [comment]
Moving to 7.0 sounds fine to me.
